backend/tests/test_ws.py

```
@@ -0,0 +1,120 @@
+        "aud": "realtoragentai",
+        "exp": now + 3600,
+        "iat": now,
+        "nbf": now,
+    }
+    return jwt.encode(payload, SECRET, algorithm=ALG)
+
+
+def test_invalid_token(client):
+    try:
+        with client.websocket_connect("/ws/agent?token=bad") as ws:
+            with pytest.raises(WebSocketDisconnect):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)